External/FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -189,6 +192,48 @@ namespace FEXCore::Context {
     return NewThreadState;
   }
 
+  bool Context::HandleSegfault(FEXCore::Core::InternalThreadState *Thread, int Signal, void *info, void *ucontext) {
+    auto CTX = Thread->CTX;
+
+    auto FaultAddress = (uintptr_t)((siginfo_t*)info)->si_addr;
+
+    bool DoFlush = false;
+
+    {
+      FHU::ScopedSignalMaskWithSharedMutex lk(CTX->MemoryEntryMutex, true);
+
+      // Get the one after FaultAddress, or end
+      // FaultAddress is inclusive
```

Comment:
It gets the one after, as the code does `--Entry;` before use.

Reply:
(if `FaultAddress` was not inclusive, we'd have to use `lower_bound` here to get the one after the address we are looking for)

Reply:
In the sense that it points /inside/ the range we want to invalidate

Reply:
Why wouldn't the range we want to invalidate include the address that triggered the fault in the first place? The wording in the comment is too ambiguous to point out something that's obvious by context.

If you want to emphasize the inside bit, how about `Get the first mapping including FaultAddress`?

Reply:
The reason it is commented is so that it contrasts with the other lookups later in the code, which do use `lower_bound`, which makes it easy to understand why one uses `lower_bound` and the other `upper_bound` when scanning for bugs.

```
// Find the first mapping at or after the range ends, or ::end().
      // Top points to the address after the end of the range
      auto Mapping = MemoryMaps.lower_bound(Top);
```

`Get the first mapping including FaultAddress` would actually not be what happens though, we Get the first mapping after FaultAddress, which is the correct thing to do, because FaultAddress will be included in the previous to that mapping, if that exists, as it is inclusive to the range we want to invalidate.

It's just a pedantic comment around the `lower_bound` / `upper_bound` usage. We've been bit by bugs around that in the past.

Reply:
> Get the first mapping including FaultAddress would actually not be what happens though, we Get the first mapping after FaultAddress, which is the correct thing to do, because FaultAddress will be included in the previous to that mapping, if that exists, as it is inclusive to the range we want to invalidate.

Point taken.

> It's just a pedantic comment around the lower_bound / upper_bound usage. We've been bit by bugs around that in the past.

I don't think this is the right way, but we're better off spending energy in one of the bigger discussions of this PR.